src/components/Modal/src/Modal.vue

```diff
@@ -7,6 +7,29 @@
 	</div>
 </template>
 
+<script>
+import modalApi from './modal-api';
+
+export default {
+	name: 'Modal',
+
+	inject: {
+		modalApi,
+	},
+
+	props: {
+		beforeClose: {
+			type: Function,
+			default: () => true,
+		},
+	},
+
+	mounted() {
+		this.modalApi.state.options.beforeCloseHook = this.beforeClose;
```

Comment:
Since `beforeClose` is a prop, it can be reactive. Instead of putting this in `mounted`, put it in a `watch` with `immediate: true`.

Reply:
Very good point, I should have done it that way to start